src/js/covidMap.js

```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
                     record: "fluid.covidMap.filterCheckboxInMap"
                 }
             }
+        },
+        desktopViewHandler: {
+            source: "{map}.model.isDesktopView",
+            type: "fluid.covidMap.desktopViewHandler"
+        },
+        mobileViewHandler: {
+            source: "{map}.model.isMobileView",
+            type: "fluid.covidMap.mobileViewHandler"
+        }
+    }
+});
+
+fluid.covidMap.detectWindowSize = function (that) {
+    const mediaQuery = window.matchMedia(that.options.mediaQueryBreakpoint);
+    that.applier.change("isDesktopView", mediaQuery.matches ? true : false);
+    mediaQuery.addListener(function (evt) {
+        that.applier.change("isDesktopView", evt.matches ? true : false);
+    });
+};
+
+// The desktop view handler
+fluid.defaults("fluid.covidMap.desktopViewHandler", {
+    gradeNames: "fluid.modelComponent",
+    listeners: {
+        "onCreate.setVisiblePanels": {
```

Comment:
For both of these viewHandlers it will be clearer to relay both visiblePanels and expandedPanels as a whole into its model, and then provide these "onCreate" arguments as initial model values. Hopefully it will work!

Reply:
@amb26, your suggestion on [FLUID-6603](https://issues.fluidproject.org/browse/FLUID-6603) makes it possible to relay the initial model values of `visiblePanelArray` and `expandedPanels` to the parent component. But I ran into another issue with these values being set incorrectly after the construction of the conditional sub-component. I've been trying to isolate the problem in order to create a test case but had no success. So I pushed up the code change in this branch and hope you can reproduce the problem.

Using the mobile sized window is easier to reproduce the problem.

**Expected initial mobile view**: The cities list is visible and collapsed. All other panels are hidden.
<img width="455" alt="expected" src="https://user-images.githubusercontent.com/572877/111691432-b140e900-8804-11eb-9aa6-66848e7a5752.png">

**Problem with relaying initial model values from the conditional sub-component**: all panels are hidden.
<img width="447" alt="problem" src="https://user-images.githubusercontent.com/572877/111691791-0bda4500-8805-11eb-82c1-911742cbc359.png">

**The finding from the debug***
1. [Tracking model changes](https://github.com/cindyli/covid-data-monitor/blob/improve-css/src/js/covidMap.js#L385-L399) on `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` shows, before the window size is detected and `mobileViewHandler` is constructed, they are set to some initial values by some other `modelRelay` actions;
2. When the window size is detected and `mobileViewHandler` is constructed, `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` are then relayed to the initial model values of `mobileViewHandler`;
3. However, when debugging `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` at the `mobileViewHandler` onCreate event, these values remain as what are set in step 1.
4. Nevertheless, as the problematic screenshot shows, none of these model values are applied so none of panels are visible.

If step 2 uses `mobileViewHandler` onCreate event to call `{map}.applier.change`, it will set `visiblePanelArray` and `expandedPanels` correctly. The debug at step 3 confirms it.

@amb26, please let me know if my steps are not clear. Thanks.

Reply:
@amb26, as we talked at today's meeting, since using "onCreate" event to set initial model value is a working work-around, I reverted to this approach from relaying the initial model values that are set directly on the `model` block.

This branch is ready for another review and merge. 

My next step is to update [the wecount website PR](https://github.com/inclusive-design/wecount.inclusivedesign.ca/pull/415) to fetch the map from this branch.